pkg/kubelet/volumemanager/reconciler/reconstruct.go

```diff
@@ -140,7 +140,7 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*gl
 	for _, gvl := range volumesNeedUpdate {
 		err := rc.actualStateOfWorld.MarkVolumeAsAttached(
 			//TODO: the devicePath might not be correct for some volume plugins: see issue #54108
-			gvl.volumeName, gvl.volumeSpec, rc.nodeName, gvl.devicePath)
+			klog.FromContext(context.Background()), gvl.volumeName, gvl.volumeSpec, rc.nodeName, gvl.devicePath)
```

Comment:
What's suspicious? That `MarkVolumeAsAttached` now needs a logger as parameter? If it does any log call, then it needs such a parameter (see also my other comment about call chain vs. component logger).

Reply:
changing `pkg/kubelet` is the suspicious part. It may be necessary (like when calling Attach() / Detach() with logger). but either the whole VolumeManager must be updated or `MarkVolumeAsAttached` should be left untouched.

Reply:
I think it is because
https://github.com/kubernetes/kubernetes/blob/6bce018b36fc26f9b002f730421866ff3e8b919e/pkg/kubelet/volumemanager/cache/actual_state_of_world.go#L47
```
ActualStateOfWorld must implement the methods required to allow operationexecutor to interact with it.
```

Reply:
`operationexecutor` is a package used both in kubelet and kube-controller-manager (KCM) I can imagine it has `logger` param only for the KCM parts. Anyway, it looks OK with the `logger` too.